kahuna/public/js/services/api/collections-api.js

```diff
@@ -64,6 +66,41 @@ apiServices.factory('collections', ['mediaApi', function (mediaApi) {
         return Promise.all(promises);
     }
 
+    function collectionsEquals(collectionsA, collectionsB) {
+        return angular.equals(
+            collectionsA.sort(),
+            collectionsB.sort()
+        );
+    }
+
+    function getCollectionsIds(imageCollections) {
+        const collectionsIds = [];
+        imageCollections.data.forEach(e => collectionsIds.push(e.pathId));
+        return collectionsIds;
+    }
+
+    function untilCollectionsEqual(image, expectedCollections) {
+        return image.get().then(apiImage => {
+            const apiCollections = imageAccessor.getCollectionsIds(apiImage);
+            if (collectionsEquals(apiCollections, expectedCollections)) {
+                return apiImage;
+            } else {
+                return $q.reject();
+            }
+        });
+    }
+
+    function removeImageFromCollection(collection, image) {
+        return collection.perform('remove')
+            .then(newImageCollections => apiPoll(() =>
```

Comment:
[It returns the new resource of collections](https://github.com/guardian/grid/blob/master/collections/app/controllers/ImageCollectionsController.scala#L54)